linkedlist.c

```diff
@@ -0,0 +1,62 @@
+#include<stdio.h>
+#include<stdlib.h>
+
+struct Node {
+	int data;
+	struct Node *next;
+};
+
+struct Node *head = NULL;
+
+void Insertion_Beginning() {
+	struct Node *newNode;
+	newNode = NULL;
+	newNode = (struct Node*)malloc(sizeof(struct Node));
+	printf("Enter data :");
+	scanf("%d", &newNode->data);
+	if (head == NULL) {
+		head = newNode;
+		head->next = NULL;
+	} else {
+		newNode->next = head;
+		head = newNode;
+	}
+	printf("\nNode inserted\n");
+}
+
+void Display(){
+	int count = 0;
+	struct Node *newNode;
+	printf("Available data :\n");
+	newNode = head;
+	while(newNode != NULL){
+		printf("%d\n", newNode->data);
+		count ++;
+		newNode = newNode->next;
+	}
+	printf("\nTotal nodes:%d\n", count);
+}
+
+int main(){
+    int choice;
+    void (*ptr [2])() = {Insertion_Beginning, Display};
+    while(1) {
+	    printf("0.Insertion_Beginning\n1.Display\n2.Exit\n");
+	    printf("Enter the choice:");
+	    scanf("%d", &choice);
+
+	    if(getchar() != '\n') {
+		    printf("\n***Enter valid integer***\n");
+		    while ((getchar()) != '\n');

```

Comment:
Why this while is required?

Reply:
This condition is only valid for non integer values. For example if input is 4.34 then scanf("%d") will read 4,remaining .34 case comes to if condition. Checks whether point(.) != '\n'. Since this is true it will print enter valid integer and also goes to a loop to read till the end(ie it reads 3 and 4 one by one) and then reaches \n and loop gets breaked.Thats why using while loop .If without while loop it will print two times (***Enter valid integer *** ) for avoid that.